tests/src/JIT/Regression/JitBlue/GitHub_4044/GitHub_4044.cs

```diff
@@ -0,0 +1,250 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Runtime.CompilerServices;
+
+class C
+{
+    public static int Main(string[] args)
+    {
+        int error = Test1();
+        error += Test2();
+        error += Test3();
+        error += Test3b();
+        error += Test4();
+        error += Test5();
+        error += Test6();
+        Console.WriteLine(error == 0 ? "Pass" : "Fail");
+        return 100 + error;
+    }
+
+    [MethodImpl(MethodImplOptions.NoInlining)]
+    static int Test1()
+    {
+        int i = 3;
+        do
+        {
+            try
+            {
+                throw new Exception();
+            }
+            // we should be decrementing i here, it does not happen
+            catch (Exception) when (--i < 0)
+            {
+                Console.Write("e");
+                break;
+            }
+            catch (Exception)
+            {
+                // just printing constant 3 here
+                //000000b5  mov         ecx,3
+                //000000ba  call        000000005B3CBAF0
+                Print1(i);
+            }
+        } while (true);
+
```

Comment:
Does this test infinite loop without your fix? Maybe Print1() should throw if (limit <= 0)?